similar/main/state.cpp

```diff
@@ -1536,6 +1536,12 @@ int state_restore_all_sub(const char *filename, const secret_restore secret)
 				do_cloak_invul_secret_stuff(old_gametime, ret_pl_info);
 			} else {
 				get_local_player() = dummy_player;
+				// Keep keys even if they died on secret level (otherwise game becomes impossible)
+				// Example: Cameron 'Stryker' Fultz's Area 51
+				pl_info.powerup_flags |= (plrobj.ctype.player_info.powerup_flags &
```

Comment:
As mentioned in the separate ticket (#314), I have no objection to changing the semantics here.  I flagged it because I thought this pull was solely about fixing regressions caused by the use of `player_info`, and this is not a regression.  Entering a secret level does not clear any keys that the player gained in the base level, so for consistency, it makes sense that returning from the secret level should not clear any keys the player gained in the secret level.

:+1: for this hunk.